.golangci.yml

```diff
@@ -40,11 +40,7 @@ linters-settings:
       - ifElseChain
       - singleCaseSwitch # Every time this occurred in the code, there  was no other way.
   revive:
-    ignore-generated-header: false
-    severity: warning
-    confidence: 0.8
-    errorCode: 1
-    warningCode: 1
+    severity: error
```

Comment:
Why was `severity` changed?

Reply:
I think it does not make sense to have a notion of warnings in any linters, either they are errors or not. We've been using similar settings for the js/css linters:

https://github.com/go-gitea/gitea/blob/fc89363832c87678d9e35e865b49c63c7ad498f2/Makefile#L376

Reply:
I don't disagree, but I also couldn't easily tell what it even _does_ when changed. 
Is it just simply "warning prints but doesn't fail, whereas error prints _and_ fails"?

I'm assuming it means the latter, which is fine with me.

Reply:
I interpret it as "all issues will report as errors". From [revive docs](https://github.com/mgechev/revive#configuration):

```
# Sets the default severity to "warning"
severity = "warning"
```

I will test if it makes any difference in golangci, but I assume not.

Reply:
Tested it, severity does not change the output of golangci-lint. The only place where it has a influence is in the output which has a default sort order:

```yaml
  sort-order:
    - linter
    - severity
    - file # filepath, line, and column.
 ```

I will change this to just `[file]` because I prefer going through lint errors by file and I think others too will prefer that ordering.